### PR TITLE
(SIMP-3324) Update puppet-agent table

### PR DESCRIPTION
**Pull request by heliocentric**
Update puppet-agent version table to include 4.10 versions

SIMP-3324 #close

### CODE REVIEW

File: files/puppet-agent-versions.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #   - https://docs.puppet.com/puppet/latest/about_agent.html
 #
 version_mappings:
+  '4.10.2': '1.10.2'
```
Comment (lnemsick-simp) [inline files/puppet-agent-versions.yaml]: Is there a reason you have left off puppet 4.10.3/puppet-agent 1.10.3?

Comment (heliocentric) [inline files/puppet-agent-versions.yaml]: nope, adding

File: lib/simp/beaker_helpers.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ module Simp; end
 
 module Simp::BeakerHelpers
   require 'simp/beaker_helpers/version'
-  DEFAULT_PUPPET_AGENT_VERSION = '1.8.3'
+  DEFAULT_PUPPET_AGENT_VERSION = '1.10.2'
```
Comment (jeefberkey) [inline lib/simp/beaker_helpers.rb]: Is this the version of the agent it installs if `PUPPET_VERSION` isn't set?

Comment (trevor-vaughan) [inline lib/simp/beaker_helpers.rb]: @jeefberkey Yeah, it is. We need to test both old and new so I guess it doesn't really matter which it is set to we just have to remember to run the old ones as well.

@heliocentric Does it make sense to set it to the oldest supported by default?

Comment (jeefberkey) [inline lib/simp/beaker_helpers.rb]: Default should be PUPPET_VERSION or the version of the puppet gem I have bundled.

**Comment by jeefberkey** [comment]
`puppet-agent` 1.10.3 and 1.10.4 are out

**Comment by jeefberkey** [comment]
Puppet 5 is out. https://docs.puppet.com/puppet/5.0/release_notes.html

File: lib/simp/beaker_helpers.rb (hunk 2 above)
Comment (nick-markowski) [inline lib/simp/beaker_helpers.rb]: @jeefberkey there's a fugly bit in the install_puppet function that gives PUPPET_VERSION hierarchy over this default. PUPPET_INSTALL_VERSION > PUPPET_VERSION > DEFAULT_PUPPET_AGENT_VERSION

Comment (trevor-vaughan) [inline lib/simp/beaker_helpers.rb]: @jeefberkey After a bit of going back and forth, it sounds like this should be whatever the *oldest* supported version by SIMP is. That way we won't start using fancy new features and have our modules die on old systems.

**Comment by trevor-vaughan** [comment]
@jeefberkey and @heliocentric This has been rebased against upstream and should be 'oldest version' compatible.

Went ahead and added a rubygems deploy key while I was in here

**Comment by trevor-vaughan** [comment]
Rebased

**Pull request merged**